File: package/ctcgfw/luci-app-wrtbwmon/luasrc/model/cbi/wrtbwmon/custom.lua
local USER_FILE_PATH = "/etc/wrtbwmon.user"

local fs = require "nixio.fs"

local f = SimpleForm("wrtbwmon", 
    translate("Usage - Custom User File"), 
    translate("This file is used to match users with MAC addresses. "
    .. "Each line must have the following format: <b><font color=\"red\">\"00:aa:bb:cc:ee:ff,username\"</font></b>."))

local o = f:field(Value, "_custom")

o.template = "cbi/tvalue"
o.rows = 20

function o.cfgvalue(self, section)
    return fs.readfile(USER_FILE_PATH)
end

function o.write(self, section, value)
    value = value:gsub("\r\n?", "\n")
    fs.writefile(USER_FILE_PATH, value)
end

f.submit = translate("Submit")

return f

File: package/ctcgfw/luci-app-wrtbwmon/htdocs/luci-static/resources/view/wrtbwmon/custom.js
'use strict';
'require fs';
'require ui';

return L.view.extend({
	load: function() {
		return fs.trimmed('/etc/wrtbwmon.user').catch(function(err) {
			ui.addNotification(null, E('p', {}, _('Unable to load the customized hostname file: ' + err.message)));
			return '';
		});
	},

	render: function(data) {
		return E('div', {'class': 'cbi-map'}, [
			E('h2', {'name': 'content'}, [ _('Usage - Custom User File') ]),
			E('div', {'class': 'cbi-map-descr'}, [
				_('Each line must have the following format:'),
				E('em', {}, E('font', {'color': 'red'}, '00:aa:bb:cc:ee:ff,hostname'))
			]),
			E('div', {'class': 'cbi-section'}, [
				E('textarea', {
					'id': 'custom_hosts',
					'style': 'width: 100%;padding: .5em;',
					'rows': 20
				}, data)
			])
		]);
	},

	handleSave: function(ev) {
		var map = document.querySelector('#custom_hosts');
		return fs.write('/etc/wrtbwmon.user', map.value.trim().replace(/\r\n/g, '\n') + '\n');
	},

	addFooter: function() {
		return E('div', { 'class': 'cbi-page-actions' }, [
			E('button', {
				'class': 'cbi-button cbi-button-save',
				'click': L.ui.createHandlerFn(this, 'handleSave')
			}, [ _('Save') ])
		]);
	}
});
